### PR TITLE
feat(app): implement insensitive operators in app

### PR DESCRIPTION
**Pull request by bernatvadell**
In relation to PR #11737 , I've implemented the rest of the operators in the application.

CC. @nordcart

### CODE REVIEW

**Comment by rijkvanzanten** [comment]
@jaycammarano Can you look into spinning up some tests for the `generate-joi` util that is touched in this PR? I'm not quite sure if we can easily test the generated Joi schema, so we might want to treat it as an integration test by providing known valid/invalid payloads against the generated schema in order to test the generation of the schema.

**Comment by jaycammarano** [comment]
> @jaycammarano Can you look into spinning up some tests for the `generate-joi` util that is touched in this PR? I'm not quite sure if we can easily test the generated Joi schema, so we might want to treat it as an integration test by providing known valid/invalid payloads against the generated schema in order to test the generation of the schema.

so we want to reorganize the shared package to have a tests folder the way we are doing with the other packages?

**Comment by rijkvanzanten** [comment]
@jaycammarano Yeah lets make that consistent 👍🏻

File: docs/getting-started/quickstart.md [hunk 1]
```diff
@@ -83,9 +83,9 @@ Once logged in, you're greeted with the option to create your first
 2. Click **"Create Collection"** and a side menu will appear.
 3. Fill in a **Name**.\
    _For the sake of this demo, we'll be calling ours `articles`, but feel free to make it your own!_
-4. Leave the other options at default. Click <span mi btn>arrow_forward</span> and the **"Optional System Fields"** menu
+4. Leave the other options at default. Click <span mi btn>arrow*forward</span> and the **"Optional System Fields"** menu
    will open.\
-   _Keep the values in this menu at the default, toggled off, for now. You can adjust them later._
+   \_Keep the values in this menu at the default, toggled off, for now. You can adjust them later.*
```
Comment (erondpowell) [inline docs/getting-started/quickstart.md:88]: @rijkvanzanten @jaycammarano 

Something weird is going on here, and i can't figure it out. Every time we make a commit, `arrow_forward` compiles to `arrow*forward`.... In the PR when I wrote this doc, I used `arrow\_forward` and that compiled into `arrow_forward` after the commit... But now, all commits appear to impact it.

Same weirdness on line 88, As you can see, it was written `_abc_` and then gets compiled to `_abc*` which is obviously subpar....

I spent at least 30 min trying to see if there was a `*` or `_` mismatch somewhere before and couldn't spot one.

Any ideas on what's up here?

**Comment by erondpowell** [comment]
@bernatvadell @jaycammarano
I smell a minor docs update coming at the end of all this :octocat: 
Please tag me when this is ready so I can make that play!
(I will try to keep up as well, but a tipoff from the main geezers is always helpful 😄 )

File: docs/getting-started/quickstart.md (hunk 1 above)
Comment (erondpowell) [reply docs/getting-started/quickstart.md:88]: Thanks @azrikahar !
Based on that bug you sent, it looks like I probably have an extra `*` somewhere....
I will track it down.

**Comment by jaycammarano** [comment]
There is a push to use `@directus/shared/utils` for generate-joi instead of the one in `app/utils`. Would you be willing to make this shift?

**Comment by rijkvanzanten** [comment]
> There is a push to use `@directus/shared/utils` for generate-joi instead of the one in `app/utils`. Would you be willing to make this shift?

This is in-works at https://github.com/directus/directus/pull/13596. Once that's cleared up, we can merge this PR 👍🏻

**Comment by nordcart** [comment]
> > There is a push to use `@directus/shared/utils` for generate-joi instead of the one in `app/utils`. Would you be willing to make this shift?
> 
> This is in-works at #13596. Once that's cleared up, we can merge this PR 👍🏻

I know that you guys have your hands tied up but #13596 will be in the next release, is it somehow possible this PR could also be in the next release?

**Comment by rijkvanzanten** [comment]
@nordcart This PR needs some tweaking to match the cleaned up structure of the generate-joi and seeing the next release is in two days, and we're a maintainer down do to covid (hope you'll feel better soon @br41nslug! 🦠), I can't promise that this'll make it in by then.

**Comment by nordcart** [comment]
> @nordcart This PR needs some tweaking to match the cleaned up structure of the generate-joi and seeing the next release is in two days, and we're a maintainer down do to covid (hope you'll feel better soon @br41nslug! 🦠), I can't promise that this'll make it in by then.

Any news on this?

**Comment by nordcart** [comment]
Maybe this could make it to the next release? :face_in_clouds:

**Comment by nordcart** [comment]
🤔

**Comment by rijkvanzanten** [comment]
Heya, thanks for your PR! Our small team has been attempting to work through a backlog of external contributions for the better part of a year, but in that time many have become too stale to merge or the change is no longer desired/relevant.. 
We've updated the [contributing guidelines](https://github.com/directus/directus/blob/main/contributing.md#pull-requests) and are working on [making more resources available](https://github.com/directus/directus/discussions/17977) to make sure that doesn’t happen again 😬 In the meantime, we'll have to close this PR for now..

**Comment by nordcart** [comment]
> Heya, thanks for your PR! Our small team has been attempting to work through a backlog of external contributions for the better part of a year, but in that time many have become too stale to merge or the change is no longer desired/relevant.. We've updated the [contributing guidelines](https://github.com/directus/directus/blob/main/contributing.md#pull-requests) and are working on [making more resources available](https://github.com/directus/directus/discussions/17977) to make sure that doesn’t happen again 😬 In the meantime, we'll have to close this PR for now..

Thank you for the info, and for the great work you do at the open source world! 
I´ve been following this for 2 years, starting with issue #5996 . 

Is there however some other way to filter o2m where only for example authors with an article "test" is shown?

Take the following minimal example:

```
authors & articles
articles.author -> authors.id

Fetch all authors that have an article where title IS NOT "Test"

The following SQL is used

SELECT
	"authors"."id",
	"authors"."name"
FROM
	"authors"
WHERE
	"authors"."id" in(
		SELECT
			"articles"."author" AS "author"
		FROM
			"articles"
		WHERE
			NOT "articles"."title" = 'Test'
	)
ORDER BY
	"authors"."id" ASC
```
However, this still fetches the top level author as soon as they have any other article that matches the search query 🤔

**Pull request closed without merging**